python/ray/data/datasource/file_based_datasource.py

```diff
@@ -607,9 +608,22 @@ def read_files(
                 rows_per_file=self._delegate._rows_per_file(),
                 file_sizes=file_sizes,
             )
-            read_task = ReadTask(
-                lambda read_paths=read_paths: read_files(read_paths, filesystem), meta
-            )
+
+            num_threads = self._delegate._NUM_THREADS_PER_TASK
```

Comment:
can we also add the check, if `_NUM_THREADS_PER_TASK <= 1`, the read should happen on main thread, without threads pool? So we can turn off the feature in case there's any issue.

Reply:
when it's 1, using `make_async_gen` is still meaningful, as read will be done in background thread. 
We can do that when it's "<= 0"